contracts/minters/open-edition-minter/src/contract.rs

```diff
@@ -599,23 +680,52 @@ pub fn execute_update_per_address_limit(
         .add_attribute("limit", per_address_limit.to_string()))
 }
 
+pub fn execute_burn_to_mint(
```

Comment:
This code is duplicated. I would import it from another contract, or move it to packages (as a controller), and import that.

Reply:
I'm not sure about this. The contract needs to call itself with a Mint Msg, which would have to match the MintMsg type expected for the contract (VendingMinter MintMsg for Vending Minter, OpenEditionMinter MintMsg for Open Edition Minter) etc. 

Or is there some way to put a generic as a parameter? 
This is the line in execute_burn_to_mint that would need a generic type in order to be packageable: 
```rs
    let execute_mint_msg = ExecuteMsg::Mint {
        token_uri: token_uri_msg.token_uri,
    };
    ```

Reply:
Marking this as unaddressable comment per Discord: 

<blockquote>
I can't make this generic. for example in the base minter there is the helpers.rs that uses generics. 
</blockquote>

```rs
use crate::msg::ExecuteMsg;

  pub fn call<T: Into<ExecuteMsg>>(&self, msg: T) -> StdResult<CosmosMsg> 
  let msg = to_binary(&msg.into())?;
```

<blockquote>
Here base minter knows how to convert into an execute msg bc it is using the crate from the same package. We can't do that if burn to mint would be in another package, it would only be able to use its own msg/execute msg. 

Need to be able to convert into execute msg so that would be able to put it into a wasm msg something like this: 
</blockquote>

```rs
        Ok(WasmMsg::Execute {
            contract_addr: self.addr().into(),
            msg,
            funds: vec![funds],
        }
 ```

<blockquote>
Then if the mint msg is generic as T it doesn't know how to convert it to binary
</blockquote>

```rs
let execute_msg = to_binary(&execute_mint_msg)?;
the trait bound `T: Serialize` is not satisfied
the trait `Serialize` is not implemented for `T`
```
<blockquote>
so even if you try to pass in the execute msg already formed into another package it won't know what to do with it
</blockquote>